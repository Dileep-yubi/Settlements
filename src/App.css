.App {
  text-align: center;
  height: 100%;
  width: 100%;
}
@import url("https://fonts.googleapis.com/css?family=Inter");
.rootContainer {
  height: 100%;
  color: #fff;
  display: flex;
  flex-direction: column;
  background: linear-gradient(168.09deg, #a067f4 0%, #7003e4 30.4%);
}
