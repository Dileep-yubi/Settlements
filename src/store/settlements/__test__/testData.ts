import { Payment } from "../../../components/BreakDown/types";

export const payments: Array<Payment> = [
  {
    title: "Tenure",
    period: '9 months'
  },
  {
    title: "EMI",
    value: 3339,
    currency: "USD",
  },
  { title: "R.O.I", interest: "1%" },
  {
    title: "Interest",
    value: 150,
    currency: "USD",
  },
  {
    title: "Total amount payable",
    value: 1000,
    currency: "USD",
  },
];
