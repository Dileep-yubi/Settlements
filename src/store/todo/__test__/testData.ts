import { Payment } from "../../../components/BreakDown/types";

export const payments: Array<Payment> = [
  {
    title: "Tenure",
    value: "9 months",
  },
  {
    title: "EMI",
    value: "3339",
    currency: "USD",
  },
  { title: "R.O.I", value: "1%" },
  {
    title: "Interest",
    value: "150",
    currency: "USD",
  },
  {
    title: "Total amount payable",
    value: "1000",
    currency: "USD",
  },
];
