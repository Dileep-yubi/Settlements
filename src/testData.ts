import { Plan } from "components/EmiPlanComponent/EmiPlanComponent";

export const headerComponent: Array<Plan> = [
  {
    header: {
      currency: "USD",
      value: 3339,
      period: "3 months",
      interest: "1%",
    },
    children: [
      {
        title: "Tenure",
        period: "4 months",
      },
      {
        title: "Interest",
        value: 25,
        currency: "USD",
      },
      {
        title: "Total amount payable",
        value: 435,
        currency: "USD",
      },
    ],
  },
  {
    header: {
      currency: "USD",
      value: 1839,
      period: "6 months",
      interest: "1.5%",
    },
    children: [
      {
        title: "Tenure",
        period: "6 months",
      },
      {
        title: "Interest",
        currency: "USD",
        value: 75,
      },
      {
        title: "Total amount payable",
        value: 10075,
        currency: "USD",
      },
    ],
  },
  {
    header: {
      currency: "USD",
      value: 1277,
      period: "9 months",
      interest: "2%",
    },
    children: [
      {
        title: "Tenure",
        period: "9 months",
      },
      {
        title: "Interest",
        value: 150,
        currency: "USD",
      },
      {
        title: "Total amount payable",
        value: 10150,
        currency: "USD",
      },
    ],
  },
];
