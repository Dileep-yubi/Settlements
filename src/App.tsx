import React, { useState } from "react";
import "./App.css";
import { EmiListComponent } from "./Components/EmiListComponent/EmiListComponent";
import { FooterComponent } from "./Components/FooterComponent";
import { Header } from "./Components/Header/Header";

const App = () => {
  const [selected, setSelected] = useState<number>(0);
  const [currentStep, setCurrentStep] = useState<number>(0);

  const steps = [
    {
      component: (
        <EmiListComponent
          selectedEmi={selected}
          handleSelected={(index) => setSelected(index)}
        />
      ),
    },
  ];

  return (
    <div className="rootContainer">
      <Header />
      {steps.map(
        (step, index) =>
          index === currentStep && (
            <React.Fragment key={index}>{step.component}</React.Fragment>
          )
      )}
      <FooterComponent
        onNext={() => setCurrentStep((prev: number) => prev + 1)}
        onPrev={() => setCurrentStep((prev: number) => prev - 1)}
        isNextDisabled={!selected || currentStep === steps.length - 1}
        isBackDisabled={!currentStep || false}
      />
    </div>
  );
};

export default App;
