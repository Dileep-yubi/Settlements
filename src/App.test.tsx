import React from "react";
import { fireEvent, render, screen } from "@testing-library/react";
import App from "./App";
import reduxMockStore from "redux-mock-store";
import { payments } from "store/settlements/__test__/testData";
import { Provider } from "react-redux";
import StyleProvider from "@yubi/yb-style-provider";
import { IntlProvider } from "react-intl";

describe("<App/>", () => {
  const mockStore = reduxMockStore(),
    store = mockStore({
      todoState: { payments: payments },
    });
  const renderComponent = () =>
    render(
      <Provider store={store}>
        <StyleProvider baseUrl={""}>
          <IntlProvider locale="en">
            <App />
          </IntlProvider>
        </StyleProvider>
      </Provider>
    );
  it("should match the snapshot", () => {
    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
    const displayTexts = [
      "SMARTBank",
      "Convert your outstanding amount into EMIs with ease!",
      "Choose EMI plan",
      "Total outstanding amount 10000",
    ];
    displayTexts.forEach((test) => {
      expect(screen.getByText(test)).toBeInTheDocument();
    });
  });

  it("should disable 'not now' and 'submit' button initially", () => {
    renderComponent();
    const submit = screen.getByRole("button", { name: "Submit" });
    expect(submit).toBeInTheDocument();
    expect(submit).toBeDisabled();
    const notNow = screen.getByRole("button", { name: "Not Now" });
    expect(notNow).toBeInTheDocument();
    expect(notNow).toBeDisabled();
  });

  it("should enable Submit button when a Emi plan is selected", () => {
    renderComponent();
    const emiPlanHeader = screen.getByText("USD 3339 for 3 months @1% p.a");
    expect(emiPlanHeader).toBeInTheDocument();
    fireEvent.click(emiPlanHeader);
    expect(emiPlanHeader).toBeInTheDocument();
    const emiPlanSummary = [
      "Tenure",
      "4 months",
      "Interest",
      "USD 25",
      "Total amount payable",
      "USD 435",
    ];
    emiPlanSummary.forEach((test) => {
      expect(screen.getByText(test)).toBeInTheDocument();
    });

    const input = screen.getByRole("radio");
    expect(input).toBeChecked();

    const submit = screen.getByRole("button", { name: "Submit" });
    expect(submit).toBeInTheDocument();
    expect(submit).toBeEnabled();
    const notNow = screen.getByRole("button", { name: "Not Now" });
    expect(notNow).toBeInTheDocument();
    expect(notNow).toBeDisabled();
  });

  it("should navigate to the EMI plan summary page when click on 'Submit' button", () => {
    renderComponent();
    const emiPlanHeader = screen.getByText("USD 3339 for 3 months @1% p.a");
    expect(emiPlanHeader).toBeInTheDocument();
    fireEvent.click(emiPlanHeader);

    const submit = screen.getByRole("button", { name: "Submit" });
    fireEvent.click(submit);

    const displayTexts = [
      "Summary",
      "Please review the selection made",
      "USD 3339 for 3 months @1% p.a",
      "View the revised EMI payment plan letter",
      "Convert your outstanding amount into EMIs with ease!",
      "Tenure",
      "4 months",
      "Interest",
      "USD 25",
      "Total amount payable",
      "USD 435",
    ];

    displayTexts.forEach((text) => {
      expect(screen.getByText(text)).toBeInTheDocument();
    });

    expect(submit).toBeInTheDocument();
    expect(submit).toBeEnabled();
    const back = screen.getByRole("button", { name: "Back" });
    expect(back).toBeInTheDocument();
    expect(back).toBeEnabled();
  });

  it("should navigate to the EMI plan Selection page when click on 'Back' button", () => {
    renderComponent();
    const emiPlanHeader = screen.getByText("USD 3339 for 3 months @1% p.a");
    expect(emiPlanHeader).toBeInTheDocument();
    fireEvent.click(emiPlanHeader);

    const submit = screen.getByRole("button", { name: "Submit" });
    fireEvent.click(submit);

    const back = screen.getByRole("button", { name: "Back" });
    fireEvent.click(back);

    expect(submit).toBeInTheDocument();
    expect(submit).toBeEnabled();
    const notNow = screen.getByRole("button", { name: "Not Now" });
    expect(notNow).toBeInTheDocument();
    expect(notNow).toBeDisabled();
    const EmiPlanHeader2 = screen.getByText("USD 3339 for 3 months @1% p.a");
    expect(EmiPlanHeader2).toBeInTheDocument();
    const emiPlanSummary = [
      "Tenure",
      "4 months",
      "Interest",
      "USD 25",
      "Total amount payable",
      "USD 435",
    ];
    emiPlanSummary.forEach((test) => {
      expect(screen.getByText(test)).toBeInTheDocument();
    });
  });
});
