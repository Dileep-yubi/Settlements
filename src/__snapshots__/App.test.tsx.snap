// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> should match the snapshot 1`] = `
<div>
  <div
    class="rootContainer"
  >
    <header
      class="headerContainer"
    >
      <h1
        class="sc-bcXHqe bFDxfp"
      >
        SMARTBank
      </h1>
      <div
        class="sc-dkrFOg iMlCA-d"
      >
        <img
          alt="Fallback"
          height="150px"
          src="https://cardinsider.com/wp-content/uploads/2021/10/slider.png.webp"
          width="50%"
        />
        <div>
          Convert your outstanding amount into EMIs with ease!
        </div>
      </div>
    </header>
    <div
      class="sc-kDvujY hiVRov"
    >
      <div
        style="padding: 1rem;"
      >
        <h1
          class="sc-bcXHqe bFDxfp"
        >
          Choose EMI plan
        </h1>
        <div
          style="padding: 20px;"
        >
          Total outstanding amount
           
          10000
        </div>
        <div
          class="flex-container"
        >
          <div
            class="emiPlanWrapper"
          >
            <div
              class="emiComponent "
            >
              <div>
                <div
                  style="padding: 6px;"
                >
                  <div
                    class="sc-dkrFOg iMlCA-d"
                  >
                    <span>
                      USD 3339 for 3 months @1% p.a
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="emiPlanWrapper"
          >
            <div
              class="emiComponent "
            >
              <div>
                <div
                  style="padding: 6px;"
                >
                  <div
                    class="sc-dkrFOg iMlCA-d"
                  >
                    <span>
                      USD 1839 for 6 months @1.5% p.a
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="emiPlanWrapper"
          >
            <div
              class="emiComponent "
            >
              <div>
                <div
                  style="padding: 6px;"
                >
                  <div
                    class="sc-dkrFOg iMlCA-d"
                  >
                    <span>
                      USD 1277 for 9 months @2% p.a
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <footer
        class="sc-hLBbgP bkuKRn"
      >
        <div
          class="sc-dkrFOg iLEIro"
        >
          <button
            class="sc-eDvSVe hAbUIn"
            disabled=""
          >
            Not Now
          </button>
          <button
            class="sc-eDvSVe hDKcWg"
            disabled=""
          >
            Submit
          </button>
        </div>
      </footer>
    </div>
  </div>
</div>
`;
